From 044b4e71c34f7d20bbee8ec29bba0a1ae2766b32 Mon Sep 17 00:00:00 2001
From: "kele.bai" <kele.bai@amlogic.com>
Date: Tue, 29 Apr 2014 10:40:34 +0800
Subject: [PATCH 3995/5965] PD #91207:enable di for interlace output.

PD #91207: modif:modify di post hold line for m8 1080i/50 output.

PD #91207: 17 fo: 17 lines for di post hold.
---
 drivers/amlogic/deinterlace/deinterlace.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/deinterlace/deinterlace.c b/drivers/amlogic/deinterlace/deinterlace.c
index e241cef85af8..2cebe2719f19 100755
--- a/drivers/amlogic/deinterlace/deinterlace.c
+++ b/drivers/amlogic/deinterlace/deinterlace.c
@@ -189,7 +189,11 @@ static unsigned char new_keep_last_frame_enable = 0;
 static int bypass_state = 1;
 static int bypass_prog = 0;
 static int bypass_hd_prog = 0;
+#if MESON_CPU_TYPE==MESON_CPU_TYPE_MESON8
+static int bypass_interlace_output = 0;
+#else
 static int bypass_interlace_output = 1;
+#endif
 #ifdef CONFIG_AM_DEINTERLACE_SD_ONLY
 static int bypass_hd = 1;
 #else
@@ -204,7 +208,11 @@ static int invert_top_bot = 0;
 static int skip_top_bot = 0;
 static char interlace_output_flag = 0;
 static int bypass_get_buf_threshold = 4;
-#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON6
+
+#if MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8
+static int post_hold_line = 17;//for m8 1080i/50 output
+static int force_update_post_reg = 0x10;
+#elif MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON6
 static int post_hold_line = 20;
 static int force_update_post_reg = 0x10;/* bit[4], 1 call process_fun every vsync; 0 call process_fun when toggle frame (early_process_fun is called)
                                         bit[3:0] set force_update_post_reg = 1: only update post reg in vsync for one time
-- 
2.19.0

